### PR TITLE
VMs methods testing

### PR DESCRIPTION
**Pull request by lehinevych**
{{pytest: cfme/tests/test_rest.py -k 'vm' -v --long-running -v}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit fe20f91658ab58a7ac03d08ac9033f121efcdc97:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/test_rest.py:1341:5: F841 local variable 'res' is assigned to but never used
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 005f78191f97fd3c833200045284341060d34e83:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e389e69ff400de1a79337433cd5abf54d571b459:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/test_rest.py [hunk 1]
```diff
@@ -1302,6 +1302,83 @@ def test_create_tag_with_wrong_arguments(rest_api):
...
+    events = db["event_streams"]
+    events_list = list(db.session.query(events).filter(
+        events.vm_name == vm,
+        events.message == event["event_message"],
+        events.event_type == event["event_type"],
+    ))
+    # Use > 1 because the vm fixture is for whole module
+    # and we are adding event twice to the same vm.
+    # So first time it should be equal 1, second > 1
+    assert(len(events_list) >= 1, "Could not find the event in the database")
```
Comment (jkrocil) [inline cfme/tests/test_rest.py]: I can see a corner case where the first time it is added correctly and events_list == 1 and second time it is not added correctly and its still 1, not 2, but the >=1 assert passes.

File: cfme/tests/test_rest.py [hunk 2]
```diff
@@ -1302,6 +1302,83 @@ def test_create_tag_with_wrong_arguments(rest_api):
...
+        pytest.skip("Set owner action for service is not implemented in this version")
+    rest_vm = rest_api.collections.vms.find_by(name=vm)[0]
+    user = rest_api.collections.users.find_by(userid='admin')[0]
+    data = {
+        "owner": {"href": user.href}
+    }
+    rest_vm.action.set_ownership(**data)
+    rest_vm.reload()
+    assert hasattr(rest_vm, "evm_owner_id")
+    assert rest_vm.evm_owner_id == user.id
```
Comment (jkrocil) [inline cfme/tests/test_rest.py]: Just curious - what is the evm_owner_id of the rest_vm at the beginning of the test?

File: cfme/tests/test_rest.py [hunk 3]
```diff
@@ -1302,6 +1302,83 @@ def test_create_tag_with_wrong_arguments(rest_api):
...
+        pytest.skip("Set owner action for service is not implemented in this version")
+    rest_vm = rest_api.collections.vms.find_by(name=vm)[0]
+    group = rest_api.collections.groups.find_by(description='EvmGroup-super_administrator')[0]
+    data = {
+        "group": {"href": group.href}
+    }
+    rest_api.collections.vms.action.set_ownership(rest_vm, **data)
+    rest_vm.reload()
+    assert hasattr(rest_vm, "miq_group_id")
+    assert rest_vm.miq_group_id == group.id
```
Comment (jkrocil) [inline cfme/tests/test_rest.py]: Same question as above - I presume there is no group owner assigned - just checking.

Comment (lehinevych) [inline cfme/tests/test_rest.py]: yes, you are right, this attribute appears after assigning.

File: cfme/tests/test_rest.py (hunk 2 above)
Comment (lehinevych) [inline cfme/tests/test_rest.py]: There is no evm_owner_id attribute in the beginning of the test.

File: cfme/tests/test_rest.py (hunk 1 above)
Comment (lehinevych) [inline cfme/tests/test_rest.py]: good point, thinking about another condition.

Comment (psav) [inline cfme/tests/test_rest.py]: Do we get an event time? I think we have a server time now that we can use to reference against?

**Comment by dajoRH** [comment]
Lint report for commit fa19cee055322b7a8424fffb5e24ddfa9ea3a525:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*
